### PR TITLE
8219567: Name of first parameter of RandomAccessFile(String,String) is inconsistent

### PR DESCRIPTION
**Pull request by bplb**
Revise some verbiage about the `RandomAccessFile(String,String)` constructor so that the string name of a file is referred to as _pathname string_ for consistency with `java.io.File`.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Change requires CSR request [JDK-8314669](https://bugs.openjdk.org/browse/JDK-8314669) to be approved
- [x] Commit message must refer to an issue

### Issues
 * [JDK-8219567](https://bugs.openjdk.org/browse/JDK-8219567): Name of first parameter of RandomAccessFile(String,String) is inconsistent (**Enhancement** - P5)
 * [JDK-8314669](https://bugs.openjdk.org/browse/JDK-8314669): Name of first parameter of RandomAccessFile(String,String) is inconsistent (**CSR**)

### Reviewers
 * [Justin Lu](https://openjdk.org/census#jlu) (@justin-curtis-lu - Committer) ⚠️ Review applies to [d8e43b53](https://git.openjdk.org/jdk/pull/15351/files/d8e43b532e35b4cdb90c590087626f6e1c5ea598)
 * [Vyom Tewari](https://openjdk.org/census#vtewari) (@vyommani - Committer) ⚠️ Review applies to [d8e43b53](https://git.openjdk.org/jdk/pull/15351/files/d8e43b532e35b4cdb90c590087626f6e1c5ea598)
 * [Roger Riggs](https://openjdk.org/census#rriggs) (@RogerRiggs - **Reviewer**)
 * [Jaikiran Pai](https://openjdk.org/census#jpai) (@jaikiran - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/15351/head:pull/15351` \
`$ git checkout pull/15351`

Update a local copy of the PR: \
`$ git checkout pull/15351` \
`$ git pull https://git.openjdk.org/jdk.git pull/15351/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 15351`

View PR using the GUI difftool: \
`$ git pr show -t 15351`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/15351.diff">https://git.openjdk.org/jdk/pull/15351.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/15351#issuecomment-1684542094)

### CODE REVIEW

**Review by justin-curtis-lu — APPROVED** [review]
LGTM

**Review by vyommani — APPROVED** [review]
Looks OK to me

File: src/java.base/share/classes/java/io/RandomAccessFile.java [hunk 1]
```diff
@@ -206,8 +206,8 @@ public RandomAccessFile(String name, String mode)
      *             not denote an existing regular file, or if the mode begins
      *             with {@code "rw"} but the given file object does not denote
      *             an existing, writable regular file and a new regular file of
-     *             that name cannot be created, or if some other error occurs
-     *             while opening or creating the file
+     *             that pathname cannot be created, or if some other error
+     *             occurs while opening or creating the file
      * @throws      SecurityException  if a security manager exists and its
      *             {@code checkRead} method denies read access to the file
      *             or the mode is {@code "rw"} and the security manager's
```
Comment (jaikiran) [inline src/java.base/share/classes/java/io/RandomAccessFile.java:213]: GitHub UI isn't allowing me to add a review comment on line 195, so adding it here. Should line 195 be changed to say:

```
also called with the pathname of the {@code file} argument to see if write access to the file...
```
to match what it says a few lines above for the `checkRead` security check?

Comment (bplb) [reply src/java.base/share/classes/java/io/RandomAccessFile.java:213]: You are correct - thanks. So changed in ef91651b35cd30a7a11ac75ebbdc03a9c29c59e3.

**Review by RogerRiggs — APPROVED** [review]
LGTM

**Review by jaikiran — APPROVED** [review]
Thank you Brian, looks good to me.

**Pull request closed without merging**